### PR TITLE
Typing: `exteinsons.util`

### PR DESCRIPTION
**Pull request by asi1024**

### CODE REVIEW

File: pytorch_pfn_extras/training/extensions/util.py [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
 import time
+from typing import NamedTuple, Optional, Sequence, TextIO, Tuple, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from pytorch_pfn_extras.training.manager import _BaseExtensionsManager
+
+
 if os.name == 'nt':
     import ctypes
+    from ctypes import windll  # type: ignore [attr-defined]
```
Comment (kmaehashi) [inline pytorch_pfn_extras/training/extensions/util.py:13]: How about this?
```suggestion
    import ctypes.windll
```

File: pytorch_pfn_extras/training/extensions/util.py [hunk 2]
```diff
@@ -15,73 +22,84 @@ class _CONSOLE_SCREEN_BUFFER_INFO(ctypes.Structure):
...
+    time: float
+
+
 class ProgressBar:
 
-    def __init__(self, out=None):
+    def __init__(self, out: Optional[TextIO] = None) -> None:
         self._out = sys.stdout if out is None else out
-        self._recent_timing = collections.deque([], maxlen=100)
+        self._recent_timing: 'queue.Queue[_QueUnit]' = queue.Queue(maxsize=100)
```
Comment (kmaehashi) [inline pytorch_pfn_extras/training/extensions/util.py:77]: Question: Why is this implementation changed?

Comment (asi1024) [reply pytorch_pfn_extras/training/extensions/util.py:77]: It is just a cosmetic change, not for a typing issue.
I can drop bb58bb88f20fb35016e6c70e0cb40d098c5d322b if it is not suitable for this pull request.

File: pytorch_pfn_extras/training/extensions/util.py (hunk 1 above)
Comment (asi1024) [reply pytorch_pfn_extras/training/extensions/util.py]: Fixed. Thanks!

**Pull request merged**